src/com/google/cose/Ec2SigningKey.java

```diff
@@ -287,18 +257,17 @@ public byte[] sign(Algorithm algorithm, byte[] message, String provider)
     verifyAlgorithmAllowedByKey(algorithm);
     verifyOperationAllowedByKey(Headers.KEY_OPERATIONS_SIGN);
 
+    ECPrivateKey key = (ECPrivateKey) keyPair.getPrivate();
     try {
-      Signature signature;
       if (provider == null) {
-        signature = Signature.getInstance(algorithm.getJavaAlgorithmId());
-      } else {
-        signature = Signature.getInstance(algorithm.getJavaAlgorithmId(), provider);
+        return new EcdsaSignJce(key, getHashType(algorithm), EcdsaEncoding.DER).sign(message);
```

Comment:
should we instead do this if we are asked to use tink provider rather than putting tink by default?